support-frontend/assets/components/orderSummary/contributionsOrderSummaryContainer.tsx

```diff
@@ -64,12 +70,45 @@ export function ContributionsOrderSummaryContainer({
 		);
 	}
 
+	const handleAmountUpdate = (updateAmountBy: number, index: number) => {
+		dispatch(
+			setSelectedAmount({
+				contributionType: contributionType,
+				amount: `${selectedAmount + updateAmountBy}`,
+			}),
+		);
+		trackComponentClick(
+			`contribution-topup-amount-${
+				updateAmountBy > 0 ? 'select' : 'deselect'
+			}-${countryGroupId}-${contributionType}-option${index}`,
+		);
+	};
+
+	const amountBeforeAmendments = useContributionsSelector(
+		getUserSelectedAmountBeforeAmendment,
+	);
+	const isWithinThreshold =
+		contributionType !== 'ONE_OFF' &&
+		amountBeforeAmendments >=
+			benefitsThresholdsByCountryGroup[countryGroupId][contributionType] &&
+		amountBeforeAmendments <=
+			checkoutTopUpUpperThresholdsByCountryGroup[countryGroupId][
+				contributionType
+			];
```

Comment:
This is extremely hard to read- any chance of refactoring this into separate named conditions?

Reply:
Yea - definitely a copy/pasta, I will have a look into this, but a little wary of changing implementation all that much.